modules/tinymce/src/themes/silver/main/ts/Autocompleter.ts

```diff
@@ -116,111 +93,69 @@ const register = (editor: Editor, sharedBackstage: UiFactoryBackstageShared) =>
+            // Use the constant.
+            'normal'
+          )
+        ),
+        {
+          anchor: {
+            type: 'node',
+            root: SugarElement.fromDom(editor.getBody()),
+            node: Optional.from(element)
+          }
         }
-      }
-    );
+      );
 
-    InlineView.getContent(autocompleter).each(Highlighting.highlightFirst);
+      InlineView.getContent(autocompleter).each(Highlighting.highlightFirst);
+    });
   };
 
-  const doLookup = (fetchOptions?: Record<string, any>): Optional<AutocompleteLookupInfo> =>
-    activeAutocompleter.get().map(
-      (ac) => getContext(editor.dom, editor.selection.getRng(), ac.triggerChar)
-        .bind((newContext) => lookupWithContext(editor, getAutocompleters, newContext, fetchOptions))
-    ).getOrThunk(() => lookup(editor, getAutocompleters));
-
-  const load = (fetchOptions?: Record<string, any>) => {
-    doLookup(fetchOptions).fold(
-      cancelIfNecessary,
-      (lookupInfo) => {
-        commenceIfNecessary(lookupInfo.context);
-
-        // Wait for the results to return and then display the menu
-        lookupInfo.lookupData.then((lookupData) => {
-          // Lookup the active autocompleter to make sure it's still active, if it isn't then do nothing
-          activeAutocompleter.get().map((ac) => {
-            const context = lookupInfo.context;
-
-            // Ensure the active autocompleter trigger matches, as the old one may have closed
-            // and a new one may have opened. If it doesn't match, then do nothing.
-            if (ac.triggerChar === context.triggerChar) {
-              const combinedItems = getCombinedItems(context.triggerChar, lookupData);
-
-              // Open the autocompleter if there are items to show
-              if (combinedItems.length > 0) {
-                display(ac, context, lookupData, combinedItems);
-                // close if we haven't found any matches in the last 10 chars
-              } else if (context.text.length - ac.matchLength >= 10) {
-                cancelIfNecessary();
-                // otherwise just hide the menu
-              } else {
-                hideIfNecessary();
-              }
-            }
-          });
-        });
-      }
-    );
-  };
+  const updateDisplay = (lookupData: AutocompleteLookupData[]) => {
+    const combinedItems = getCombinedItems(lookupData);
 
-  const onKeypress = Throttler.last((e) => {
-    // IE will pass the escape key here, so just don't do anything on escape
-    if (e.which === 27) {
-      return;
+    // Open the autocompleter if there are items to show
+    if (combinedItems.length > 0) {
+      display(lookupData, combinedItems);
+    } else {
+      hideIfNecessary();
     }
+  };
+
+  editor.on('AutocompleterStart', ({ lookupData }) => {
+    activeState.set(true);
+    processingAction.set(false);
+    updateDisplay(lookupData);
+  });
+
+  editor.on('AutocompleterUpdate', ({ lookupData }) => updateDisplay(lookupData));
```

Comment:
I just realised there may be a potential bug here... if the combined items from the update is 0 then it's going to hide the autocompleter but not update the state cells. As such `hideIfNecessary` probable should be setting `processingAction` and `activeState` to `false` to protect against that.

Reply:
Actually scratch that sorry, I was getting confused with the old `cancelIfNecessary` sorry.